dealer/src/OkexPerpetualSwapStrategy.ts

```diff
@@ -525,75 +525,109 @@ export class OkexPerpetualSwapStrategy implements HedgingStrategy {
             }
             await database.internalTransfers.insert(intTransferRecord)
 
-            // then initiate the withdrawal which by default uses the funding account
-            const config = this.exchangeConfig as OkexExchangeConfiguration
-            const averageFee = config.minOnChainWithdrawalFee
-            const withdrawArgs = {
-              currency: TradeCurrency.BTC,
-              quantity: transferSizeInBtc - averageFee,
-              address: withdrawOnChainAddress,
-              params: {
-                fee: averageFee,
-                dest: DestinationAddressType.External,
-                pwd: this.exchange.fundingPassword,
-              },
-            }
-            const withdrawalResult = await this.exchange.withdraw(withdrawArgs)
+            // only withdraw if funding balance > MINIMUM_FUNDING_BALANCE_BTC
+            //  and if so, withdraw (funding balance - MINIMUM_FUNDING_BALANCE_BTC)
+            //  i.e. newTransferSizeInBtc = max(0, funding balance - MINIMUM_FUNDING_BALANCE_BTC)
+            let newTransferSizeInBtc = transferSizeInBtc
+            const fundingAccountBalanceResult = await this.getFundingAccountBalance()
             this.logger.debug(
-              { withdrawArgs, withdrawalResult },
-              "withdraw() returned: {withdrawalResult}",
+              { fundingAccountBalanceResult },
+              "getFundingAccountBalance() returned: {fundingAccountBalanceResult}",
             )
-
-            // keep record of the attempt
-            const extTransferRecord: ExternalTransfer = {
-              isDepositNotWithdrawal: false,
-              currency: withdrawArgs.currency,
-              quantity: withdrawArgs.quantity,
-              destinationAddressTypeId: withdrawArgs.params.dest,
-              toAddress: withdrawArgs.address,
-              fundPassword: withdrawArgs.params.pwd,
-              fee: Number(withdrawArgs.params.fee),
-              chain: SupportedChain.BTC_Bitcoin,
-              // transferId: null,
-              success: false,
-            }
-
-            if (!withdrawalResult.ok) {
-              await database.externalTransfers.insert(extTransferRecord)
-              return { ok: false, error: withdrawalResult.error }
+            if (fundingAccountBalanceResult.ok) {
+              newTransferSizeInBtc = Math.max(
+                0,
+                fundingAccountBalanceResult.value.btcFreeBalance -
+                  hedgingBounds.MINIMUM_FUNDING_BALANCE_BTC,
```

Comment:
any reason to just not do a comparison?  `fundingAccountBalanceResult.value.btcFreeBalance > hedgingBounds.MINIMUM_FUNDING_BALANCE_BTC`